File: frontend/src/api/gamification.ts
import { api } from "@/api/client";
import type { Badge, LeaderboardEntry } from "@/types";

export async function getBadges() {
  const { data } = await api.get<{ badges: Badge[] }>("/badges");
  return data.badges;
}

export async function getLeaderboard() {
  const { data } = await api.get<{ leaderboard: LeaderboardEntry[] }>("/leaderboard");
  return data.leaderboard;
}

File: backend/app/schemas/gamification.py
from pydantic import BaseModel


class Badge(BaseModel):
    name: str
    description: str
    unlocked: bool


class BadgeListResponse(BaseModel):
    badges: list[Badge]


class LeaderboardEntry(BaseModel):
    name: str
    role: str
    points: int
    progress: int


class LeaderboardResponse(BaseModel):
    leaderboard: list[LeaderboardEntry]
